tooling/lsp/src/requests/mod.rs

```diff
@@ -20,25 +21,60 @@ use crate::{
 // They are not attached to the `NargoLspService` struct so they can be unit tested with only `LspState`
 // and params passed in.
 
+mod code_lens_request;
 mod goto_definition;
 mod profile_run;
 mod test_run;
 mod tests;
 
 pub(crate) use {
+    code_lens_request::collect_lenses_for_package, code_lens_request::on_code_lens_request,
     goto_definition::on_goto_definition_request, profile_run::on_profile_run_request,
     test_run::on_test_run_request, tests::on_tests_request,
 };
 
+/// LSP client will send initialization request after the server has started.
+/// [InitializeParams].`initialization_options` will contain the options sent from the client.
+#[derive(Debug, Deserialize, Serialize)]
+struct LspInitializationOptions {
+    /// Controls whether code lens is enabled by the server
+    /// By default this will be set to true (enabled).
+    #[serde(rename = "enableCodeLens", default = "default_enable_code_lens")]
+    enable_code_lens: bool,
+}
+
+fn default_enable_code_lens() -> bool {
+    true
+}
+
+impl Default for LspInitializationOptions {
+    fn default() -> Self {
+        Self { enable_code_lens: default_enable_code_lens() }
+    }
+}
+
 pub(crate) fn on_initialize(
     state: &mut LspState,
     params: InitializeParams,
 ) -> impl Future<Output = Result<InitializeResult, ResponseError>> {
     state.root_path = params.root_uri.and_then(|root_uri| root_uri.to_file_path().ok());
 
-    async {
+    let initialization_options = match params.initialization_options {
```

Comment:
unwrap_or_default ?

Reply:
Not sure I follow here. `.unwrap_or_default` should be equivalent.